okhttp/src/main/java/okhttp3/internal/connection/Transmitter.java

```diff
@@ -123,6 +125,8 @@ Exchange newExchange(Interceptor.Chain chain, boolean doExtensiveHealthChecks) {
 
     synchronized (connectionPool) {
       this.exchange = result;
+      this.exchangeRequestDone = false;
+      this.exchangeResponseDone = false;
```

Comment:
those two are needed when redirection happens, because they have to be bound to the exchange lifecycle, not the transmitter.